analysis_driver/quality_control/contamination_blast.py

```diff
@@ -0,0 +1,107 @@
+        blast_outfile = fasta_file.split('.')[0] + '_blastn'
+        blast_cmd = "%s -query %s -db %s -out %s -outfmt '6 qseqid sseqid length pident evalue sgi sacc staxids sscinames scomnames stitle'" % (blastn_bin, fasta_file, nt_db, blast_outfile)
+        return blast_cmd, blast_outfile
+
+    def retrieve_identified_taxa(self, blast):
+        taxids = {}
+        classes = {}
+        ncbi = NCBITaxa()
+
+        with open(blast) as openfile:
+            blast = openfile.readlines()
+            for line in blast:
+                taxid = line.split()[7]
+                if taxid not in taxids:
+                    taxids[taxid] = 1
+                else:
+                    taxids[taxid] += 1
+
+        for taxon in taxids:
+            num_reads = taxids[taxon]
+            lineage = ncbi.get_lineage(taxon)
+            names = ncbi.get_taxid_translator(lineage)
+            rank = ncbi.get_rank(lineage)
+            class_taxid = [i for i in rank if rank[i] == 'family']
+            if class_taxid:
+                class_name = names[class_taxid[0]]
+                if class_name not in classes:
+                    classes[class_name] = num_reads
+                else:
+                    classes[class_name]+=num_reads
+        return classes
+
+    def run_sample_fastq(self, fastq_file):
+        self.dataset.start_stage('sample_fastq')
+        sample_fastq_command, fasta_outfile = self.sample_fastq_command(fastq_file)
+        sample_fastq_executor = executor.execute(
+            sample_fastq_command,
+            job_name='sample_fastq',
+            working_dir=self.working_dir,
+            cpus=2,
+            mem=10
+        )
+        exit_status = sample_fastq_executor.join()
+        self.dataset.end_stage('sample_fastq', exit_status)
+        return fasta_outfile
+
+    def run_blast(self, fasta_file):
+        self.dataset.start_stage('contamination_blast')
+        fasta_blast_command, blast_outfile = self.fasta_blast_command(fasta_file)
+        contamination_blast_executor = executor.execute(
+            fasta_blast_command,
+            job_name='contamination_blast',
+            working_dir=self.working_dir,
+            cpus=2,
+            mem=10
+        )
+        exit_status = contamination_blast_executor.join()
+        self.dataset.end_stage('contamination_blast', exit_status)
+        return blast_outfile
+
+    def check_for_contamination(self):
+        taxa_identified = {}
+        for fastq_file in self.fastq_files:
+
+            fasta_outfile = self.run_sample_fastq(fastq_file)
+            blast_outfile = self.run_blast(fasta_outfile)
+            taxa = self.retrieve_identified_taxa(blast_outfile)
+
+            for taxon in taxa:
+                if taxon not in taxa_identified:
+                    taxa_identified[taxon] = taxa[taxon]
+                else:
+                    taxa_identified[taxon] += taxa[taxon]
+        return taxa_identified
```

Comment:
I think those taxa identified ought to be written to file. maybe in some kind of json format.